core/src/main/java/com/linecorp/armeria/internal/RequestContextUtil.java

```diff
@@ -87,5 +87,20 @@ public static SafeCloseable pushWithRootAndOldCtx(ClientRequestContext currentCt
         }
     }
 
+    /**
+     * An {@link IllegalStateException} which is raised when pushing a context from the unexpected thread
+     * or forgetting to close the previous context.
+     */
+    public static final class IllegalContextPushingException extends IllegalStateException {
```

Comment:
Could you make it a top-level class and make it part of `common` package?

Reply:
So If I change just to return the `IllegalStateException` from the previous method, I don't have to declare this new exception type. And I think we don't have to have this another exception, especially for this case. (Sorry I made this class because I misunderstood the comment. 🙏) Do you think we still need this?

Reply:
Nope. Let's continue using `IllegalStateException`.